Calculation develop bot segregation refactoring v1

Y-Pazniak opened the pull request:
Fixed structure fail with wrong relationships between higher and lower classes. 
Fixed naming in laguage resource.

Y-Pazniak committed:
Segregation was successful. Bot works alone. Nobody doesn't know about it except main class.

Y-Pazniak committed:
Language versions naming was fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/Main.java:
```diff
@@ -9,12 +10,22 @@
         }
     }
+
+    private static void singletonInits() {
```
I'm not sure that it makes sense.
We will init classes in Main but in a different way.

Y-Pazniak replied:
Removed it. Waiting for instructions to fix it. 
To be honest, when I did this, I clearly understood that I was doing useless thing. But I had to do something to try init Singletons in the main class, but I hadn't another ideas how to handle this request except this silly way.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/MessageRouter.java:
```diff
@@ -1,22 +1,25 @@
 package by.custom.utilcalculator.service;
```
Please, move it under `controller` package
```suggestion
package by.custom.utilcalculator.controller;
```

Y-Pazniak replied:
Done.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java:
```diff
@@ -0,0 +1,100 @@
+        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+    }
+
+    public SendMessage setCarOriginFieldAndGetNextMessageForUser(Message inputMessage) {
```
I prefer that BotFieldsManager doesn't know about telegram-specific classes, `SendMessage`, `Message`.
```suggestion
    public String setCarOriginFieldAndGetNextMessageForUser(User sender, command: Commands) {
```

Y-Pazniak replied:
I fixed it, but in a little another way. Now BotFieldsManager doesn't know anything about telegram-specific classes. 
Your suggestion is: 
public String setCarOriginFieldAndGetNextMessageForUser**(User sender, command: Commands)** {
My variant is: 
public String setCarOriginFieldAndGetNextMessageForUser(**String command**) {
I am not sure if  I am right. But in my opinion we don't need to add here anything about sender - we can add it in the MessageRouter class. And we don't need to give array: we send only one command. In my opinion my variant is better, because we send less options into the methods. But probably I made a mistake. Please, check it.

MaximPilipeiko replied:
You are absolutely right. But finally we need user info to save user progress. But it's another story.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/directory/resources/Commands.java:
It's more common practice to use singular name for enum

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/directory/resources/Prices.java:
```diff
@@ -1,4 +1,4 @@
-package by.custom.utilcalculator.repository.resources;
+package by.custom.utilcalculator.directory.resources;
 
 public class Prices {
```
It's more common practice to use singular name for enum 
```suggestion
public class Price {
```

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java:
```diff
@@ -0,0 +1,100 @@
+        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+    }
+
+    public SendMessage setCarOriginFieldAndGetNextMessageForUser(Message inputMessage) {
```
I think in this case we can use a simpler but enough informative method name. The same for other methods.
```suggestion
    public String processCarOrigin(String inputMessage) {
```

Y-Pazniak replied:
Fixed. 
There are two operations in this methods. First operation - to change the field of Bot object. 
`case Command.EAES -> botEntity.setCountryOrigin(CountryOrigin.EAES);`
The second operation is to return String with message to send user like an answer.
`return messagesManager.createSendMessage();`
Is it OK or we must to split this method in two different methods?

MaximPilipeiko replied:
It's ok for service. It's a right place to aggregate these things, store progress, and prepare answer.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/directory/BotEntity.java:
```diff
@@ -0,0 +1,63 @@
+
+import by.custom.utilcalculator.directory.steps.*;
+
+public class BotEntity {
```
Maybe `UserProgress` will be more informative.

Y-Pazniak replied:
Done.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java:
```diff
@@ -0,0 +1,100 @@
+            case Commands.JURIDICAL_PERSON -> botEntity.setOwnersType(OwnersType.JURIDICAL);
+            case Commands.PHYSICAL_PERSON -> botEntity.setOwnersType(OwnersType.PHYSICAL);
+        }
+        return messagesManager.createSendMessage(inputMessage.getChatId().toString(),
```
Please, move `createSendMessage(` to MessageRouter

Y-Pazniak replied:
I think, we don't need this method anymore. I have removed it at all. Now we can call MessagesManagers methods from BotFieldsManager class. We don't need this wrapper createSendMessage() anymore, I guess. All it's logic is in the MessageRouter router and MessageManager. So I removed it.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/MessagesManager.java:
```diff
@@ -1,131 +1,133 @@
 import org.telegram.telegrambots.meta.api.objects.Message;
 
-public class MessagesCreator {
+public class MessagesManager {
```
Now I think we have a lot of managers and `MessageCreator` reflects a sense better.

Y-Pazniak replied:
Renamed it to MessagesCreator

Y-Pazniak committed:
Singleton calling in main was fixed.
MessageRouter removed to the controller package.
BotFieldsManager doesn't know about telegram-specific classes.
Singular name for enum added.
Simpler but more informative methods names added.
BotEntity renamed to UserProgress.

Y-Pazniak committed:
public String createSendMessage() was removed from MessageRouter

Y-Pazniak committed:
Mistake with properties file was fixed

Y-Pazniak committed:
Mistake with properties file was fixed

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:
```diff
@@ -0,0 +1,72 @@
+
+    public SendMessage checkMessageBeforeExecutionAndGetResult(Update update) {
+        Message message = update.getMessage();
+        if (message.hasText()) {
```
Just it looks nicer. 
```suggestion
        if (!message.hasText()) {
            SendMessage sorrySendMessage = new SendMessage();
            sorrySendMessage.setChatId(update.getMessage().getChatId().toString());
            sorrySendMessage.setText(getSorryMessage());
            return sorrySendMessage;
        }
        
        return getCheckInputMessageAndGetAnswer(message);
```

Y-Pazniak replied:
Yes, it is really shorter than my variant. Fixed, thanks.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:
```diff
@@ -0,0 +1,72 @@
+                    sendMessage.setText(botFieldsManager.processEngineVolume(usersMessage));
+            default -> sendMessage.setText(getSorryMessage());
+        }
+        return sendMessage;
```
I prefer do not split creating object on several lines.
```suggestion
        return new SendMessage(userId, answer);
```

Y-Pazniak replied:
Sorry, it is impossible to add userId and answer into SendMessage object - it's constructor is empty. 
We can write a wrapper, but in my opinion it is reduntant. 
I fixed it in other way: 
```
        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(userID);
        sendMessage.setText(answer);
        return sendMessage;
```

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java:
```diff
@@ -0,0 +1,93 @@
+
+public class BotFieldsManager {
+    private final UserProgress userProgress;
+    private final MessagesCreator messagesManager;
```
```suggestion
    private final MessagesCreator messagesCreator;
```

Y-Pazniak replied:
Fixed.

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/service/BotFieldsManager.java:
```diff
@@ -0,0 +1,93 @@
+import by.custom.utilcalculator.directory.resources.Command;
+import by.custom.utilcalculator.directory.steps.*;
+
+public class BotFieldsManager {
```
It's not call to action
```suggestion
public class UserAnswerManager {
```

Y-Pazniak replied:
Renamed to UserProgressManager

MaximPilipeiko commented on src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:
```diff
@@ -0,0 +1,72 @@
+        return MessagesCheckerHolder.MESSAGES_CHECKER;
+    }
+
+    public SendMessage checkMessageBeforeExecutionAndGetResult(Update update) {
```
It's not call to action
```suggestion
    public SendMessage route(Update update) {
```

Y-Pazniak replied:
Renamed.
